scout/server/extensions/beacon_extension.py

```diff
@@ -0,0 +1,192 @@
+        Returns:
+            datasets(list): list of dataset _ids. Example: ["cust002_GRCh37", "cust000_GRCh38", ..]
+        """
+        datasets = []
+        json_resp = get_request_json(url=f"{self.beacon_url}/")
+        if json_resp.get("status_code") == 200:
+            datasets = json_resp.get("content", {}).get("datasets", [])
+        else:
+            flash("Error retrieving Beacon's dataset list:{json_resp}")
+        return [dset["id"] for dset in datasets]
+
+    def base_submission_data(self, store, case_obj, form):
+        """Create data dictionary to be send as json data in a POST request to the beacon "add" endpoint
+
+        Args:
+            store(adapter.MongoAdapter)
+            case_obj(dict): scout.models.Case
+            form(ImmutableMultiDict): request form submitted by user. Example:
+                [('case', 'internal_id'), ('samples', 'affected'), ('vcf_files', 'vcf_snv'), ('vcf_files', 'vcf_snv_research'), ('panels', '6246b25121d86882e127710c')]
+
+        Returns:
+            data(dict): a dictionary with base info to be use as json data in beacon add request (lacks path to VCF file to extract variants from)
+        """
+        # Initialize key/values to be sent in request:
+        assembly = "GRCh38" if "38" in str(case_obj.get("genome_build", "37")) else "GRCh37"
+        dataset_id = "_".join([case_obj["owner"], case_obj.get("build", assembly)])
+
+        samples = []
+        if form.get("samples") == "affected":
+            samples = [
+                ind["individual_id"]
+                for ind in case_obj.get("individuals", [])
+                if ind["phenotype"] == 2
+            ]
+        else:
+            samples = [ind["individual_id"] for ind in case_obj.get("individuals")]
+
+        data = {
+            "dataset_id": dataset_id,
+            "samples": samples,
+            "assemblyId": assembly,
+        }
+
+        gene_ids = set()
+        for panel in form.getlist("panels"):
+            gene_ids.update(store.panel_to_genes(panel_id=panel, gene_format="hgnc_id"))
+
+        if gene_ids:
+            data["genes"] = {"ids": list(gene_ids), "id_type": "HGNC"}
+
+        return data
+
+    def add_variants(self, store, case_obj, form):
+        """Adding variants from one of more individuals of case to Beacon
+
+        Args:
+            store(adapter.MongoAdapter)
+            case_obj(dict): scout.models.Case
+            form(ImmutableMultiDict): request form submitted by user. Example:
+                [('case', 'internal_id'), ('samples', 'affected'), ('vcf_files', 'vcf_snv'), ('vcf_files', 'vcf_snv_research'), ('panels', '6246b25121d86882e127710c')]
+
+        """
+
+        # Check if user has rights to submit case to beacon
+        user_obj = store.user(current_user.email)
+
+        if "beacon_submitter" not in user_obj.get("roles", []):
+            flash("You don't have permission to use the Beacon tool", "warning")
```

Comment:
Also something like please write to support and ask your institute primary contact to approve you as a Beacon submitter?

Reply:
Yes, good idea!